dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -179,7 +179,7 @@ def stage_extras
       flash[:info] = I18n.t(:stage_extras_teacher_message).html_safe
     end
 
-    if params[:section_id]
+    if current_user&.teacher? && params[:section_id]
       @section = current_user.sections.find_by(id: params[:section_id])
```

Comment:
this line was failing when `current_user` was nil, so protect against this by checking for `current_user` (+ the additional guard to make sure the user is actually a teacher -- we don't need to check `current_user.sections` for students)